bob/src/cluster/operations.rs

```diff
@@ -1,19 +1,20 @@
 use crate::link_manager::LinkManager;
 use crate::prelude::*;
-use super::support_types::RemoteDeleteError;
+use super::support_types::{RemoteDeleteError, RemotePutResponse, RemotePutError};
 
-pub(crate) type Tasks<Err> = FuturesUnordered<JoinHandle<Result<NodeOutput<()>, NodeOutput<Err>>>>;
+pub(crate) type Tasks<Res, Err> = FuturesUnordered<JoinHandle<Result<NodeOutput<Res>, NodeOutput<Err>>>>;
 
 // ======================= Helpers =================
 
-fn is_result_successful<TErr: Debug>(
-    join_res: Result<Result<NodeOutput<()>, NodeOutput<TErr>>, JoinError>,
+fn is_result_successful<TRes, TErr: Debug>(
+    join_res: Result<Result<NodeOutput<TRes>, NodeOutput<TErr>>, JoinError>,
     errors: &mut Vec<NodeOutput<TErr>>,
+    get_affected_count: impl Fn(TRes) -> usize
```

Comment:
Instead of passing a function as parameter, it would be better to declare a trait, that has function returning the number of affected replicas, and add restriction that `TRes` should support that trait. 
After that you can implement that trait on `NodeOutput<()>` by returning `1` and on `NodeOutput<RemotePutResponse>` returning the actual affected replicas count